src/Neuron/Zettelkasten.hs

```diff
@@ -150,14 +159,32 @@ generateSite writeHtmlRoute' zettelsPat = do
 
 -- | Create a new zettel file and return its slug
 -- TODO: refactor this
-newZettelFile :: Path b Dir -> Text -> IO String
-newZettelFile inputDir ztitle = do
+newZettelFile :: Path b Dir -> NewCommand -> IO ()
+newZettelFile inputDir NewCommand{..} = do
   zId <- Z.zettelNextIdForToday inputDir
   zettelFileName <- parseRelFile $ toString $ Z.zettelIDSourceFileName zId
   let srcPath = inputDir </> zettelFileName
   doesFileExist srcPath >>= \case
     True ->
       fail $ "File already exists: " <> show srcPath
     False -> do
-      writeFile (toFilePath srcPath) $ "---\ntitle: " <> toString ztitle <> "\n---\n\n"
-      pure $ toFilePath srcPath
+      writeFile (toFilePath srcPath) $ "---\ntitle: " <> toString title <> "\n---\n\n"
+
+      let path = toFilePath srcPath
+
+      if edit
+        then do
+          maybeEditor <- Env.getEnv "EDITOR"
+
+          editor <- case maybeEditor of
+            Nothing -> do
+              putStrLn "Set the EDITOR environment variable"
+
+              Exit.exitFailure
```

Comment:
[Q] maybe the path of the new fine should still be printedv

Reply:
I agree; it should print the new file path regardless. If the user gets this error, and they don't see the path to the created filepath, they wouldn't easily know how to open it manually.